### ISSUE TITLE
Triple-slash directive fails to parse

### ISSUE THREAD

**Issue by gwhitney** [DanielXMoore/Civet#684]
See https://civet.dev/playground/?code=Ly8vIDxyZWZlcmVuY2UgcGF0aD0iLi4vbm9kZV9tb2R1bGVzL0B0eXBlcy9tb28vaW5kZXguZC50cyIgLz4KbW9vIGZyb20gLi4vbm9kZV9tb2R1bGVzL21vby9tb28uanMK
(code is
```
/// <reference path="../node_modules/@types/moo/index.d.ts" />
moo from ../node_modules/moo/moo.js
```
If you remove the first line, all is well.

(And yes, I know this is not a great way to do this, but I was having some resolution issues so I thought I would try everything completely explicit first, and then work on the resolution.)

**Comment by STRd6** [comment]
This is because in Civet `///` is used to start a block RegEx (not yet documented on the site sadly).

We could probably add support for `///` directives (if people actually still use them) since they are only valid on the first line of the file.

**Comment by gwhitney** [comment]
Well yes it is true that I would not be trying to use triple slash directives if I weren't in civet -> tsc -> node module ES6/UMD/CommonJs import/resolution hell and trying everything I can think of to get each step in the pipeline to be happy. Once I do, I will surely find a way not to have to use it. So I am not really advocating that it has to be supported, it was just a surprising non-consistency with Typescript since, well, I figured that any line that starts with // is a comment, so surely Civet won't touch that (so right, as a matter of language design, do you _really_ want to adopt a syntax that messes with the usual dramatic convention of "literally _anything_ can come after the comment starter except the comment ender without affecting my program"? But that's a side issue to the one I raised here...) Thanks for the heads up.

**Comment by STRd6** [comment]
Civet is a complex language that inherits from many sources so there is sometimes surprising behavior. We got `///` from CoffeeScript and `//`,`/*` from TS/JS and we're generally able to get them to play nice together. We can usually do the right thing based on context.

Thanks for opening the issue about TypeScript's `///` directives, we should be able to support them.

I'm sorry for your esm/cjs nightmare. We've all been there 😿

**Comment by STRd6** [comment]
Published fix in v0.6.29
